src/ceph-daemon

```diff
@@ -1131,20 +1130,39 @@ def command_shell():
         mounts[pathify(args.config)] = '/etc/ceph/ceph.conf:z'
     if args.keyring:
         mounts[pathify(args.keyring)] = '/etc/ceph/ceph.keyring:z'
+    podman_args = ['--privileged']
+    if args.command:
+        command = args.command
+    else:
+        command = ['bash']
+        podman_args += [
+            '-it',
+            '--env', 'LANG=C',
+        ]
     c = CephContainer(
         image=args.image,
         entrypoint='doesnotmatter',
         args=[],
-        podman_args=['--privileged'],
+        podman_args=podman_args,
         volume_mounts=mounts)
     return subprocess.call(c.shell_cmd(args.command))
```

Comment:
Minor fixup for `shell` cmd when no args are passed -

```suggestion
     return subprocess.call(c.shell_cmd(command))
```

```
```
$ sudo ceph-daemon shell
Traceback (most recent call last):
  File "./sage2", line 1683, in <module>
    r = args.func()
  File "./sage2", line 1145, in command_shell
    return subprocess.call(c.shell_cmd(args.command))
  File "./sage2", line 714, in shell_cmd
    ] + cmd[1:]
IndexError: list index out of range
```
```

Reply:
sigh, fixed.  I want to do something like
```
diff --git a/qa/standalone/test_ceph_daemon.sh b/qa/standalone/test_ceph_daemon.sh
index 69ecb242646..af17925f1c1 100755
--- a/qa/standalone/test_ceph_daemon.sh
+++ b/qa/standalone/test_ceph_daemon.sh
@@ -133,6 +133,7 @@ $SUDO $CEPH_DAEMON unit --fsid $FSID --name mon.a -- is-enabled
 ## shell
 $SUDO $CEPH_DAEMON --image $IMAGE shell -- true
 $SUDO $CEPH_DAEMON --image $IMAGE shell --fsid $FSID -- test -d /var/log/ceph
+echo 'whoami ; exit' | $SUDO $CEPH_DAEMON --image $IMAGE shell
 
 ## enter
 expect_false $SUDO $CEPH_DAEMON enter
@@ -142,6 +143,8 @@ $SUDO $CEPH_DAEMON enter --fsid $FSID --name mon.a -- pidof ceph-mon
 expect_false $SUDO $CEPH_DAEMON enter --fsid $FSID --name mgr.x -- pidof ceph-mon
 $SUDO $CEPH_DAEMON enter --fsid $FSID --name mgr.x -- pidof ceph-mgr
 
+echo 'pidof ceph-mgr ; exit' | $SUDO $CEPH_DAEMON enter --fsid $FSID --name mgr.x
+
 ## ceph-volume
 $SUDO $CEPH_DAEMON --image $IMAGE ceph-volume --fsid $FSID -- inventory --format=json \
       | jq '.[]'
```
but i have a feeling this needs to use something like expect since there is maybe no tty?

Reply:
Or maybe remove the `-t` option since a pty is not in use?

```
diff --git a/src/ceph-daemon b/src/ceph-daemon
index 949af8ffb2..c698ae152b 100755
--- a/src/ceph-daemon
+++ b/src/ceph-daemon
@@ -1138,7 +1138,7 @@ def command_shell():
     else:
         command = ['bash']
         podman_args += [
-            '-it',
+            '-i',
             '--env', 'LANG=C',
         ]
     c = CephContainer(
@@ -1159,7 +1159,7 @@ def command_enter():
     else:
         command = ['bash']
         podman_args += [
-            '-it',
+            '-i',
             '--env', 'LANG=C',
         ]
     c = get_container(args.fsid, daemon_type, daemon_id,
```